### ISSUE TITLE
Invalid relaxed DKIM body hash if the message ens with mixed whitespace and newlines

### ISSUE THREAD

**Issue by andris9** [postalsys/mailauth#17]
DKIM relaxed body hash calculation should trim all whitespace and newlines from the end of the email. Instead it trims starting from the last line that has at least one whitespace byte

```
Header\r\n
\r\n
Trims correctly\r\n
\r\n
\r\n
\r\n

-->

Header\r\n
\r\n
Trims correctly\r\n
```

```
Header\r\n
\r\n
Does not trim correctly\r\n
\r\n
\t\r\n
\r\n

-->

Header\r\n
\r\n
Does not trim correctly\r\n
\r\n
```

**Comment by andris9** [comment]
The body hash calculation operates on a stream of bytes, it does not buffer the entire body to memory, so it can not use regular string manipulation functions.

**Comment by andris9** [comment]
Fixed in v2.3.4
